### PR TITLE
[Improvements]: add json signatures to cosmos signing output

### LINKED ISSUE #2566: Improvement: API update for TW_Cosmos_Proto_SigningOutput
When using JSON signing mode for cosmos, an example of the output we receive is:
```
{
  "mode": "block",
  "tx": {
    "fee": {
      "amount": [
        {
          "amount": "0",
          "denom": "uosmo"
        }
      ],
      "gas": "250000"
    },
    "memo": "",
    "msg": [
      {
        "type": "osmosis/gamm/swap-exact-amount-in",
        "value": {
          "routes": [
            {
              "poolId": "1",
              "tokenOutDenom": "ibc/27394FB092D2ECCD56123C74F36E4C1F926001CEADA9CA97EA622B25F41E5EB2"
            }
          ],
          "sender": "osmo15jetnnj03usvx4naz2lzdw7vxysu2g43546dvf",
          "tokenIn": {
            "amount": "100000",
            "denom": "uosmo"
          },
          "tokenOutMinAmount": "8873"
        }
      }
    ],
    "signatures": [
      {
        "pub_key": {
          "type": "tendermint/PubKeySecp256k1",
          "value": "AzI5lC/562WeXfc7mCAL5fy2u9CopbOeRiJbbnXpIC4m"
        },
        "signature": "+S8kqrU31NKBDTrTzIIjsrj5xynrWaUER+TTNG33ZCEIucYPp47eZWc2JmbFewF7vKFPq23QBLEZXAiQijEQ/A=="
      }
    ]
  }
}
```

Currently we have two properties: `output.json` will return the whole JSON and `output.signature` will return the signature itself (`+S8kqrU31NKBDTrTzIIjsrj5xynrWaUER+TTNG33ZCEIucYPp47eZWc2JmbFewF7vKFPq23QBLEZXAiQijEQ/A==`).

DApps expect the full signature format (`signatures` property) and currently we have to construct it ourselves on the client:
https://github.com/trustwallet/trust-web3-provider/blob/0697a61211e671b894b0c47e6d1ff28cedac084e/ios/TrustWeb3Provider/DAppWebViewController.swift#L731-L737

Would be nice if we can have it built directly from wallet-core.

### PR DESCRIPTION
**Pull request by Milerius**
## Description

fix #2566

## How to test

Run C++ unit tests

## Types of changes

Improvements for cosmos signing output

## Checklist

- [x] Create pull request as draft initially, unless its complete.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.
- [x] If there is a related Issue, mention it in the description.

If you're adding a new blockchain

- [x] I have read the [guidelines](https://developer.trustwallet.com/wallet-core/newblockchain#integration-criteria) for adding a new blockchain.

### CODE REVIEW

File: tests/Evmos/SignerTests.cpp [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+                                                        {
+                                                            "type":"ethermint/PubKeyEthSecp256k1",
+                                                            "value":"AlcobsPzfTNVe7uqAAsndErJAjqplnyudaGB0f+R+p3F"
+                                                        },
+                                                "signature":"RWt8aaBxdMAeEjym8toWskJ6WaJpEF9Ciucz2lAHkvNnTicGpzxwTUzJbJXRirSnGkejhISaYtDw2RBiq0vg5w=="
+                                            }
+                                        ]}
+                            })"_json;
+
+    /// This tx is not broadcasted, we just want to test the signature format (ethermint/PubKeyEthSecp256k1)
```
Comment (hewigovens) [inline tests/Evmos/SignerTests.cpp:71]: can you try broadcasting it? I tried last time and got signature error

**Comment by hewigovens** [comment]
Add `string signatureJson = 5;` to `SigningOutput` in `Cosmos.proto`

**Comment by rsrbk** [comment]
Thanks for helping. Could you trigger a new wallet-core release after it's merged?

File: src/Cosmos/JsonSerialization.cpp [hunk 2]
```diff
@@ -183,10 +184,10 @@ static json messagesJSON(const Proto::SigningInput& input) {
     return j;
 }
 
-static json signatureJSON(const Data& signature, const Data& pubkey) {
+json signatureJSON(const Data& signature, const Data& pubkey, TWCoinType coin) {
     return {
         {"pub_key", {
-            {"type", TYPE_PREFIX_PUBLIC_KEY},
+            {"type", coin == TWCoinType::TWCoinTypeNativeEvmos ? TYPE_EVMOS_PREFIX_PUBLIC_KEY : TYPE_PREFIX_PUBLIC_KEY},
```
Comment (hewigovens) [inline src/Cosmos/JsonSerialization.cpp:190]: can move this to another helper function?

**Review by hewigovens — APPROVED** [review]

File: tests/Evmos/SignerTests.cpp (hunk 1 above)
Comment (Milerius) [reply tests/Evmos/SignerTests.cpp:71]: will do in another pr - see #2571

File: src/Cosmos/Signer.cpp [hunk 3]
```diff
@@ -51,10 +53,13 @@ Proto::SigningOutput Signer::signProtobuf(const Proto::SigningInput& input, TWCo
 
         auto output = Proto::SigningOutput();
         const std::string jsonSerialized = buildProtoTxJson(input, serializedTxRaw);
+        auto publicKey = PrivateKey(input.private_key()).getPublicKey(TWPublicKeyTypeSECP256k1);
+        auto signatures = nlohmann::json::array({signatureJSON(signature, Data(publicKey.bytes), coin)});
```
Comment (catenocrypt) [inline src/Cosmos/Signer.cpp:57]:
```suggestion
        auto signatures = nlohmann::json::array({signatureJSON(signature, publicKey.bytes, coin)});
```
No need to copy Data to Data agian :)

File: src/proto/Cosmos.proto [hunk 4]
```diff
@@ -318,4 +318,8 @@ message SigningOutput {
 
     // Set in case of error
     string error = 4;
+
+
```
Comment (catenocrypt) [inline src/proto/Cosmos.proto:322]:
```suggestion

```

**Review by catenocrypt — APPROVED** [review]
LGTM

**Review by hewigovens — DISMISSED** [review]

**Review by miloserdow — APPROVED** [review]

**Review by hewigovens — APPROVED** [review]

**Pull request merged**